### PR TITLE
Login & signup function

### PR DESCRIPTION
**Pull request by TTBoiMike**
- created apiClient for communication with express api
- added in protectedrouting that will redirect users depending on if they are logged in
- added a signup function to create an account (error handling needs looking at)
- added a login function to log into account by storing token in localstorage
- miscellaneous styling and layout changes

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,46 @@
...
+    window.localStorage.setItem("UserToken", userToken);
+  }
+  
+  let logOutFunc = () => {
+    window.localStorage.removeItem("UserToken");
+    isLoggedInHandler(false)
+  }
+
+  // uncomment line 31 to logout current use
+  window.localStorage.removeItem("UserToken")
```
Comment (matthewcotton) [inline src/App.js:30]: This line is not commented out. Should it be?

File: src/components/Entry/EntryTabs.js [hunk 2]
```diff
@@ -1,49 +1,105 @@
...
+            apiClient.signUp(username, password)
+                .then(response => {
+                    toastr.success(response.data.message)
+                    setTabShow("left")
+                    e.target.username.value = ""
+                    e.target.password.value = ""
+                    e.target.passwordCheck.value = ""
+                })
+                .catch(err => {
+                    alert(err) // look at handling errors better - require a response to output message
```
Comment (matthewcotton) [inline src/components/Entry/EntryTabs.js:37]: I suggest using a toastr here as that is what is being used elsewhere for user notifications

File: src/components/Entry/EntryTabs.js [hunk 3]
```diff
@@ -1,49 +1,105 @@
...
+        e.preventDefault();
+        let username = e.target.username.value;
+        let password = e.target.password.value;
+        setButtonDisabled({ login: true })
+        apiClient.logIn(username, password)
+            .then(response => {
+                logInFunc(response.data.token)
+            })
+            .catch(err => {
+                // setFormError({ logIn: err })
```
Comment (matthewcotton) [inline src/components/Entry/EntryTabs.js:54]: Why is this commented out?

File: src/components/Entry/EntryTabs.js [hunk 4]
```diff
@@ -1,49 +1,105 @@
 import React, { useState } from 'react'
```
Comment (matthewcotton) [inline src/components/Entry/EntryTabs.js:1]: I suggest reformatting this document

File: src/App.js [hunk 5]
```diff
@@ -1,24 +1,46 @@
...
   return (
-    <div className="App">
+    <div className="App py-5">
       <Router>
```
Comment (matthewcotton) [inline src/App.js:33]: We need to add an unprotected route for the welcome / intro page

**Review by matthewcotton — COMMENTED** [review]
After signing up an empty toastr message appears and it is not clear if the signup process was successful. Although it was as I can login with the newly created credentials.

File: src/App.js (hunk 1 above)
Comment (TTBoiMike) [reply src/App.js:30]: this is essentially the logout function without having a logout function - when it is uncommented you can log in and then uncomment to log out. i used it so i could test the login function

File: src/components/Entry/EntryTabs.js (hunk 2 above)
Comment (TTBoiMike) [reply src/components/Entry/EntryTabs.js:37]: this alert was for production value only - how erorrs are handled needs to be looked at but i planned to use toastr with the final product

File: src/App.js (hunk 5 above)
Comment (TTBoiMike) [reply src/App.js:33]: unprotected route for login/signup page is on line 35 - although this route does redirect to dashboard if user is logged in so they can't login more than once

Comment (TTBoiMike) [reply src/App.js:33]: the welcome page is something i have forgetton to do to honest  - so i can look at that as part of a seperate issue

File: src/components/Entry/EntryTabs.js (hunk 3 above)
Comment (TTBoiMike) [reply src/components/Entry/EntryTabs.js:54]: again this is due to error handling which we said we'd look at - this is how i want to handle errors by setting them as a message to output using state

Comment (TTBoiMike) [reply src/components/Entry/EntryTabs.js:54]: again this is due to error handling which we we'll look at - this is how i want to handle errors by setting them as a message to output using state

**Pull request merged**